client/my-sites/upgrades/domain-search/site-redirect.jsx

```diff
@@ -1,63 +1,82 @@
 /**
  * External dependencies
  */
-var page = require( 'page' ),
-	React = require( 'react' );
+import page from 'page';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var HeaderCake = require( 'components/header-cake' ),
-	Main = require( 'components/main' ),
-	SiteRedirectStep = require( './site-redirect-step' ),
-	observe = require( 'lib/mixins/data-observe' );
+import HeaderCake from 'components/header-cake';
+import Main from 'components/main';
+import SiteRedirectStep from './site-redirect-step';
+import { isSiteUpgradeable } from 'state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import QueryProductsList from 'components/data/query-products-list';
 
-var SiteRedirect = React.createClass( {
-	mixins: [ observe( 'productsList', 'sites' ) ],
-
-	propTypes: {
+class SiteRedirect extends Component {
+	static propTypes = {
+		cart: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.object,
+		isSiteUpgradeable: React.PropTypes.bool,
 		productsList: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired
-	},
-
-	componentWillMount: function() {
-		this.checkSiteIsUpgradeable();
-	},
+		translate: React.PropTypes.func.isRequired,
+	};
 
-	componentDidMount: function() {
-		this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
-	},
+	constructor() {
+		super();
+		this.handleBackToDomainSearch = this.handleBackToDomainSearch.bind( this );
+	}
 
-	componentWillUnmount: function() {
-		this.props.sites.off( 'change', this.checkSiteIsUpgradeable );
-	},
+	componentDidMount() {
+		this.checkSiteIsUpgradeable();
+	}
 
-	checkSiteIsUpgradeable: function( ) {
-		var selectedSite = this.props.sites.getSelectedSite();
+	componentWillReceiveProps() {
+		this.checkSiteIsUpgradeable();
+	}
 
-		if ( selectedSite && ! selectedSite.isUpgradeable() ) {
+	checkSiteIsUpgradeable() {
+		if ( this.props.selectedSite && ! this.props.isSiteUpgradeable ) {
 			page.redirect( '/domains/add' );
 		}
-	},
+	}
 
-	backToDomainSearch: function() {
-		page( '/domains/add/' + this.props.sites.getSelectedSite().slug );
-	},
+	handleBackToDomainSearch() {
+		page( '/domains/add/' + this.props.selectedSite.slug );
```

Comment:
Does it make sense to alter this to use the `getSelectedSiteSlug` selector for this one, so it'll be slightly more resistant to possible future structural changes to the `site` object?